pkg/storage/storagewrappers/context.go

```diff
@@ -9,57 +9,67 @@ import (
 	"github.com/openfga/openfga/pkg/storage"
 )
 
-// ContextTracerWrapper is a wrapper around a datastore that passes a new
-// context to the underlying datastore methods.
-// This is so that if the original context gets cancelled (e.g by the client), the underlying database connection isn't closed.
-// So, we let outstanding queries run their course even if the original context gets cancelled to avoid database connection churning.
+// ContextTracerWrapper is a wrapper for a datastore that introduces a new context to the underlying datastore methods.
+// Its purpose is to prevent the closure of the underlying database connection in case the original context is cancelled,
+// such as when a client cancels the context. This ensures that ongoing queries are allowed to complete even if the
+// original context is cancelled, helping to avoid unnecessary database connection churn.
 //
-// ContextTracerWrapper must be the first wrapper around the datastore if traces are to work properly.
+// It is crucial for ContextTracerWrapper to be the first wrapper around the datastore for traces to function correctly.
 type ContextTracerWrapper struct {
 	storage.OpenFGADatastore
 }
 
 var _ storage.OpenFGADatastore = (*ContextTracerWrapper)(nil)
 
+// NewContextWrapper creates a new instance of ContextTracerWrapper, wrapping the specified datastore.
 func NewContextWrapper(inner storage.OpenFGADatastore) *ContextTracerWrapper {
 	return &ContextTracerWrapper{inner}
 }
 
-// queryContext returns a new context (not a child context) with a timeout and
-// the same span data as the supplied context.
+// queryContext generates a new context, independent of the provided context,
+// with a timeout, and inherits the span data from the given context.
 func queryContext(ctx context.Context) context.Context {
 	span := trace.SpanFromContext(ctx)
 	return trace.ContextWithSpan(context.Background(), span)
 }
 
+// Close ensures proper cleanup and closure of resources associated with the OpenFGADatastore.
 func (c *ContextTracerWrapper) Close() {
 	c.OpenFGADatastore.Close()
 }
 
+// Read reads data from the underlying OpenFGADatastore.
```

Comment:
Let's not duplicate docs here. All the methods here are already documented in `storage`. We should do just

```
// ReadUserTuple please see [[storage.RelationshipTupleReader.ReadUserTuple]]
```

Reply:
I agree but with one caveat. _If_ we're going to document these publicly exported member functions, then we should be diligent about documenting their differences with the more general godoc in the `storage` interface package. For example, within the context of the `ContextTraceWrapper` various methods override the provided context and so they do not share cancellation with upstream callers. That is worth noting in the documentation if we're going to document it, but otherwise we should just avoid the duplication.

Reply:
This doesn't seem to work with godoc:

<img width="1002" alt="image" src="https://github.com/openfga/openfga/assets/28300158/745f2773-5c64-43c3-97a9-72808edec6f2">

Reply:
Oh sorry, can you try with one just angle bracket? See https://go.dev/doc/comment#:~:text=The%20square%20brackets%20in%20%5Bpath/filepath%5D%20create%20a%20documentation%20link.

Reply:
Update: to note that this  `[storage.RelationshipTupleReader.ReadUserTuple]` won't take us directly to the `ReadUserTuple` function within the `RelationshipTupleReader` interface both in an IDE (Goland) or godoc generation, because there is no anchor `#RelationshipTupleReader.ReadPage` inside the generated HTML. So what I opted to do is `[storage.RelationshipTupleReader].ReadUserTuple` so we end up at least with the anchor `#RelationshipTupleReader `, which exists.